### PR TITLE
feat: guix on tags, improve caching

### PR DESCRIPTION
**Pull request by PastaPastaPasta**
## Issue being fixed or feature implemented
See commits

## What was done?
auto do guix untags, and improve caching from run to run

## How Has This Been Tested?
hasn't

## Checklist:
  _Go over all the following points, and put an `x` in all the boxes that apply._
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone _(for repository code-owners and collaborators only)_

### CODE REVIEW

File: .github/workflows/guix-build.yml [hunk 1]
```diff
@@ -46,16 +46,21 @@ jobs:
...
           path: |
             ${{ github.workspace }}/.cache
             ${{ github.workspace }}/dash/depends/built
             ${{ github.workspace }}/dash/depends/sources
             ${{ github.workspace }}/dash/depends/work
-          key: ${{ runner.os }}-guix
+          key: ${{ runner.os }}-guix-${{ github.event.pull_request.head.sha }}
+          restore-keys: |
+            ${{ runner.os }}-guix-${{ github.event.pull_request.head.sha }}
+            ${{ runner.os }}-guix-
```
Comment (knst) [inline .github/workflows/guix-build.yml:61]: is it typo, that `...-guix-` tag is over by a hyphen?

Comment (PastaPastaPasta) [reply .github/workflows/guix-build.yml:61]: wym?

Comment (knst) [reply .github/workflows/guix-build.yml:61]:
```
            ${{ runner.os }}-guix-${{ github.event.pull_request.head.sha }}
            ${{ runner.os }}-guix- <-------------------
```
it looks like copy-paste error. It seems as not full line, but cut in the middle.

If that's full line and it's a mask, why does it need a full key above then?

Comment (PastaPastaPasta) [reply .github/workflows/guix-build.yml:61]: no, it's intended, so the key is what it sets; which includes the commit id, and then the next ones are the "restore keys" so first we try the exact match, ie the one with the commit id, and then after that we try to match against one without the commit id.

**Comment by DashCoreAutoGuix** [comment]
Guix Automation has began to build this PR tagged as v21.0.0-devpr6166.7bbe3cbd. A new comment will be made when the image is pushed.

**Comment by DashCoreAutoGuix** [comment]
Guix Automation has completed; a release should be present here: https://github.com/dashpay/dash-dev-branches/releases/tag/v21.0.0-devpr6166.7bbe3cbd. The image should be on dockerhub soon.

**Pull request closed without merging**